libs/libc/misc/lib_mutex.c

```diff
@@ -389,22 +249,17 @@ int nxmutex_clocklock(FAR mutex_t *mutex, clockid_t clockid,
 
   /* Wait until we get the lock or until the timeout expires */
 
-  do
+  if (abstime)
```

Comment:
```
 * Description:
 *   This function attempts to lock the mutex referenced by 'mutex'.  If the
 *   mutex value is (<=) zero, then the calling task will not return until it
* successfully acquires the lock or timed out
```
I think removing the loop violates this promise, either one has to be fixed.

Reply:
Seems like neither -EINTR nor -ECANCELED is a valid return value from any mutex_lock variant, as pthread_mutex_lock is not a cancellation point. So can -ECANCELED get through still ? If it does this is a POSIX violation.